src/v/iceberg/rest_client/catalog_client.h

```diff
@@ -121,6 +134,7 @@ class catalog_client {
     std::reference_wrapper<client_source> _client_source;
     ss::sstring _endpoint;
     credentials _credentials;
+    ss::abort_source& _as;
```

Comment:
Since the catalog client is short lived, it accepts an abort source from the caller to know when to shut down.